class-components/src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+
+interface PaginationProps {
+  currentPage: number;
+  itemsPerPage: number;
+  totalItems: number;
+  onPageChange: (page: number) => void;
+}
+
+const Pagination: React.FC<PaginationProps> = ({
+  currentPage,
+  itemsPerPage,
+  totalItems,
+  onPageChange,
+}) => {
+  const totalPages = Math.ceil(totalItems / itemsPerPage);
+
+  const handleClick = (page: number) => {
+    onPageChange(page);
+  };
+
+  const renderPageNumbers = () => {
+    const pageNumbers = [];
+    for (let i = 1; i <= totalPages; i++) {
+      pageNumbers.push(
+        <button
+          key={i}
+          onClick={() => handleClick(i)}
+          className={currentPage === i ? 'active' : ''}
+        >
+          {i}
+        </button>
+      );
+    }
+    return pageNumbers;
+  };
```

Comment:
Why are you pushing html elements into an array? :O It's quite a heavy entity to store in an array without a particular reason

If you want to create an array of page numbers, just do 
```
function getPageNumbers(totalPages: number) {
 const pageNumbers = [];
    for (let i = 1; i <= totalPages; i++) {
      pageNumbers.push(i);
   }

return pageNumbers ;
}
```

`const pageNumbers  = getPageNumbers(totalPages);`

And then render them in your component like this (no need for a separate function really, but you can create it if you wish):
```
{pageNumbers.map(number => (
<button
          key={i}
          onClick={() => handleClick(i)}
          className={currentPage === i ? 'active' : ''}
        >
          {i}
        </button>
)}
```